NewDawn/NewDawn/ProfileViewController/Profile_DraftFinal.swift

```diff
@@ -28,6 +28,10 @@ class Profile_DraftFinal: UIViewController {
         self.navigationController?.setNavigationBarHidden(true, animated: false)
     }
     
+    override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
```

Comment:
You should remove the segue from the button to the main tab view controller. Instead, add another segue from profile draft viewer controller itself to the main tab viewer controller. By doing this, the button won't perform segue automatically. And when you call performSegue, the actual segue can be activated. And then you don't need this override anymore.